diffkemp/utils.py

```diff
@@ -19,3 +19,35 @@ def get_llvm_version():
     """
     return int(subprocess.check_output(
         ["llvm-config", "--version"]).decode().rstrip().split(".")[0])
+
+class EndLineError(Exception):
+    """
+    Error to inform that end line of function / type was not found.
+    """
+    
+    pass
+
+def get_end_line(filename, start, kind):
+    """
+    Get end line of function / type.
+    Can raise UnicodeDecodeError, EndLineError.
+    """
+    
+    if kind == "function":
+        terminator_list = ["}", ");"]
+    elif kind == "type":
+        terminator_list = ["};"]
+    
+    with open(filename, "r", encoding='utf-8') as file:
+        lines = file.readlines()
+
+        # The end of the function is detected as a line that contains
+        # nothing but an ending curly bracket
+        line_index = start - 1
+        line = lines[line_index]
+        while line.rstrip() not in terminator_list:
+            line_index += 1
+            if line_index == len(lines):
+                raise EndLineError
+            line = lines[line_index]
+        return line_index + 1
```

Comment:
I think if it returns `line_index` it will return *index* of the line where the function ends. I originally meant it to return a *number* of the line where the function ends. I probably named the function badly.

Reply:
Ah, right. Ok, since it's a utility function which may possibly be used elsewhere, it's reasonable to stick with line numbers. Maybe it'd be best to use line numbers also in the `syntax_diff` function and just use `lines[line - 1]` to access them.